src/map/unit.cpp

```diff
@@ -3132,32 +3132,44 @@ int unit_counttargeted(struct block_list* bl)
  * @param src Current target
  * @param target New target
  **/
-int unit_changetarget(struct block_list *bl, va_list ap) {
-	struct unit_data *ud = unit_bl2ud(bl);
-	struct block_list *src = va_arg(ap,struct block_list *);
-	struct block_list *target = va_arg(ap,struct block_list *);
+int unit_changetarget(block_list *bl, va_list ap) {
+	if (bl == nullptr)
+		return 1;
+	unit_data *ud = unit_bl2ud(bl);
+	block_list *src = va_arg(ap, block_list *);
+	block_list *target = va_arg(ap, block_list *);
 
-	if (!ud || !target || ud->target == target->id)
+	if (ud == nullptr || src == nullptr || target == nullptr || ud->target == target->id)
 		return 1;
-	if (!ud->target && !ud->target_to)
+	if (ud->target <= 0 && ud->target_to <= 0)
 		return 1;
 	if (ud->target != src->id && ud->target_to != src->id)
 		return 1;
 
-	if (bl->type == BL_MOB)
-		(BL_CAST(BL_MOB,bl))->target_id = target->id;
-	if (ud->target_to)
-		ud->target_to = target->id;
-	else
-		ud->target_to = 0;
-	if (ud->skilltarget)
-		ud->skilltarget = target->id;
-	unit_set_target(ud, target->id);
+	unit_change_target(*ud, target->id);
 
 	//unit_attack(bl, target->id, ud->state.attack_continue);
 	return 0;
 }
 
+/**
+ * Changes the target of a unit
+ * @param ud: Unit data
+ * @param target_id: New target ID
+ **/
+void unit_change_target(unit_data& ud, int target_id) {
+	if (target_id <= 0 || status_isdead(map_id2bl(target_id)))
+		return;
```

Comment:
```suggestion
	if (status_isdead(target))
		return;
```
You need to merge the master branch and then apply this suggestion.